scripts/nbconverted/4_similarity_analysis.py

```diff
@@ -0,0 +1,173 @@
+# 
+# Briefly, SVCCA uses Singular Value Decomposition (SVD) to extract the components explaining 99% of the variation. This is done to remove potential dimensions described by noise. Next, SVCCA performs a Canonical Correlation Analysis (CCA) on the SVD matrices to identify maximum correlations of linear combinations of both input matrices. The algorithm will identify the canonical correlations of highest magnitude across and within algorithms of the same dimensionality.
+
+# In[1]:
+
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+import os
+import pandas as pd
+import numpy as np
+import random
+import glob
+import umap
+
+from ggplot import *
+from functions import cca_core
+from sklearn.decomposition import PCA
+from numpy.random import seed
+randomState = 123
+seed(randomState)
+
+
+# In[2]:
+
+
+# Parameters
+analysis_name = 'treatment'
+num_batches = [1,2,3,4,5,6,7,8,9,10,15,20,25,30,35,40,45,50]
+
+
+# In[3]:
+
+
+# Load data
+batch_dir = os.path.join(
+    os.path.dirname(os.getcwd()),
+    "data",
+    "batch_simulated",
+    analysis_name)
+
+
+# In[4]:
+
+
+# Calculate Similarity
+
+output_list = []
+
+for i in num_batches:
+    print(i)
+    if i ==1:
+        batch_data_file = os.path.join(
+            batch_dir,
+            "Batch_"+str(i)+".txt")
+
+        batch_data = pd.read_table(
+            batch_data_file,
+            header=0,
+            sep='\t',
+            index_col=0)
+
+        # UMAP embedding of decoded batch data
+        original_data_UMAPencoded = umap.UMAP().fit_transform(batch_data)
+        original_data_UMAPencoded_df = pd.DataFrame(data=original_data_UMAPencoded,
+                                                 index=batch_data.index,
+                                                 columns=['1','2'])
+    batch_file = os.path.join(
+        batch_dir,
+        "Batch_"+str(i)+".txt")
+
+    batch_data = pd.read_table(
+        batch_data_file,
+        header=0,
+        sep='\t',
+        index_col=0)
+
+    # UMAP embedding of decoded batch data
+    batch_data_UMAPencoded = umap.UMAP().fit_transform(batch_data)
+    batch_data_UMAPencoded_df = pd.DataFrame(data=batch_data_UMAPencoded,
+                                             index=batch_data.index,
+                                             columns=['1','2'])
+
+    # SVCCA
+    svcca_results = cca_core.get_cca_similarity(original_data_UMAPencoded_df.T,
+                                          batch_data_UMAPencoded_df.T,
+                                          verbose=True)
+    
+    output_list.append(np.mean(svcca_results["cca_coef1"]))
+
+# Convert output to pandas dataframe
+svcca_umap_df = pd.DataFrame(output_list, columns=["svcca_mean_similarity"], index=num_batches)
+svcca_umap_df
+
```

Comment:
they look almost identical with small, almost random variation. is it possible svcca is not the right tool for this?

Reply:
or perhaps try simulation from raw data rather than normalized data?

Reply:
I may be misunderstanding your idea here, but we're using normalized data because the VAE needs to have each feature (gene) to be in the same range to train.

Reply:
as we discussed, this may be a byproduct of the generator not working properly so nullifying my comment here.